Scripts/download-portal-item-data.swift

```diff
@@ -76,47 +90,70 @@ func nameOfFileInArchive(at url: URL) throws -> String {
     return String(data: filenameData, encoding: .utf8)!.trimmingCharacters(in: .whitespacesAndNewlines)
 }
 
-/// Uncompresses the data in the archive at the source URL into the destination
-/// URL.
+func countFilesInArchive(at url: URL) throws -> Int {
+    let outputPipe = Pipe()
+    let process = Process()
+    process.executableURL = URL(fileURLWithPath: "/usr/bin/zipinfo", isDirectory: false)
+    process.arguments = ["-t", url.path]
+    process.standardOutput = outputPipe
+    try process.run()
+    process.waitUntilExit()
+    
+    // The totals info looks like "240 files, 29461066 bytes uncompressed, 28292775 bytes compressed:  4.0%"
+    let totalsInfo = outputPipe.fileHandleForReading.readDataToEndOfFile()
+    // Extract the count from the info string
+    let totalsCount = String(data: totalsInfo, encoding: .utf8)!.components(separatedBy: " ")[0]
+    return Int(totalsCount)!
+}
+
+/// Uncompresses the data in the archive at the source URL into the destination URL.
 ///
 /// - Parameters:
 ///   - sourceURL: The URL of a ZIP archive.
 ///   - destinationURL: The URL at which to uncompress the archive.
 func uncompressArchive(at sourceURL: URL, to destinationURL: URL) throws {
-    FileManager.default.createFile(atPath: destinationURL.path, contents: nil)
-    let fileHandle = try FileHandle(forWritingTo: destinationURL)
+    let outputPipe = Pipe()
     let process = Process()
     process.executableURL = URL(fileURLWithPath: "/usr/bin/unzip", isDirectory: false)
-    process.arguments = ["-p", sourceURL.path]
-    process.standardOutput = fileHandle
+    // Unzip the archive into the specified sub-folder.
+    process.arguments = [sourceURL.path, "-d", destinationURL.path]
```

Comment:
It is unfortunate that the files in `PacificSouthWest` are in a directory. It would be nice to get that changed, but for now, maybe we could pass the `-j` option? That may need to be changed in the future, but it looks like it would work for now.

Reply:
Yes, `-j` would work for now, but we might need to set up a standard on how archives should be structured, to avoid the folder within subfolder situation.